lib/std/setutils.nim

```diff
@@ -35,3 +38,28 @@ template toSet*(iter: untyped): untyped =
   for x in iter:
     incl(result, x)
   result
+
+macro enmRange(enm: typed): untyped = result = newNimNode(nnkCurly).add(enm.getType[1][1..^1])
+
+template fullSet*(T: typedesc): untyped =
```

Comment:
Any specific reason for making this a template and not just a proc?

Reply:
Tired programming resulting in me not reaching the desired goal of inserting a constant.

Reply:
what do you mean, can you explain?

this seems to work:

`proc fullSet*(T: typedesc): auto {.inline.} =`

Reply:
Well prior to the current state i wasnt returning a constant I was iterating to make the constant.